track_analyzing/utils.cpp

```diff
@@ -0,0 +1,91 @@
+#include "track_analyzing/utils.hpp"
+
+#include "track_analyzing/serialization.hpp"
+
+#include "routing/segment.hpp"
+
+#include <cstdint>
+
+using namespace routing;
+using namespace std;
+
+namespace track_analyzing
+{
+double CalcSubtrackLength(MatchedTrack::const_iterator begin, MatchedTrack::const_iterator end,
+                          Geometry & geometry)
+{
+  double length = 0.0;
+
+  Segment prevSegment;
+  for (auto it = begin; it != end; ++it)
+  {
+    MatchedTrackPoint const & point = *it;
+    Segment const & segment = point.GetSegment();
+    if (segment != prevSegment)
+    {
+      length += MercatorBounds::DistanceOnEarth(geometry.GetPoint(segment.GetRoadPoint(false)),
+                                                geometry.GetPoint(segment.GetRoadPoint(true)));
+      prevSegment = segment;
+    }
+  }
+
+  return length;
+}
+
+double CalcTrackLength(MatchedTrack const & track, Geometry & geometry)
+{
+  return CalcSubtrackLength(track.begin(), track.end(), geometry);
+}
+
+double CalcSpeedKMpH(double meters, uint64_t secondsElapsed)
+{
+  CHECK_GREATER(secondsElapsed, 0, ());
+  double constexpr kMPS2KMPH = 60.0 * 60.0 / 1000.0;
+  return kMPS2KMPH * meters / static_cast<double>(secondsElapsed);
+}
+
+bool IsFiltered(string const & argument, string const & variable)
```

Comment:
The name is misleading. Could you please create a class, say,
```c++
class StringFilter
{
public:
  StringFilter() = default;  /* no filtering */
  explicit StringFilter(string const & filter);  /* filter by |filter| */;

  bool Matches(string const & s);
};
```
and use it. Really, the name and the signature are misleading.